src/Component/AbstractComponent.php

```diff
@@ -120,28 +121,11 @@ public function transformData (
 			}
 		}
 
-		$previewData = null;
-
-		if (\is_string($data["_editable"] ?? null) && \preg_match('~^<!--#storyblok#(.*)-->$~', $data["_editable"], $matches))
-		{
-			try
-			{
-				$previewData = \json_decode(\stripslashes($matches[1]), true, flags: \JSON_THROW_ON_ERROR);
-			}
-			catch (\JsonException $exception)
-			{
-				throw new InvalidDataException(\sprintf(
-					"Encountered invalid preview data: '%s'",
-					$data["_editable"],
-				), previous: $exception);
-			}
-		}
-
 		$componentData = new ComponentData(
 			$data["_uid"],
 			static::getKey(),
 			$transformedData,
-			previewData: $previewData,
+			previewData: PreviewDataParser::parse($data["_editable"] ?? null),
```

Comment:
We should just pass the untouched data through and defer the actual parsing to the normalizer.

Then we can inline the `PreviewDataParser::parse` as a method in `ComponentPreviewData`.